Bunkum.Protocols.Http/Socket/SocketHttpListenerContext.cs

```diff
@@ -12,9 +12,9 @@ public SocketHttpListenerContext(System.Net.Sockets.Socket socket, Stream stream
     protected override async Task SendResponseInternal(HttpStatusCode code, ArraySegment<byte>? data = null)
     {
         // this is dumb and stupid
-        this.ResponseHeaders.Add("Server", "Bunkum");
-        this.ResponseHeaders.Add("Connection", "close");
-        this.ResponseHeaders.Add("Date", DateTime.UtcNow.ToString("ddd, dd MMM yyyy HH:mm:ss 'GMT'"));
+        this.ResponseHeaders.TryAdd("Server", "Bunkum");
+        this.ResponseHeaders.TryAdd("Connection", "close");
+        this.ResponseHeaders.TryAdd("Date", DateTime.UtcNow.ToString("ddd, dd MMM yyyy HH:mm:ss 'GMT'"));
```

Comment:
Why does this need to change?

Reply:
if its `Add` instead of `TryAdd`, if you add these manually beforehand, it causes an error here, mainly an issue with `Server`, if you want to change that header in your responses, you cant right now
`Connection` and `Date` *probably* dont need the change though

Reply:
I don't follow why any of these would be changed in the first place but I suppose I can allow it :sweat_smile:

Reply:
according to [the docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Server), `The Server header describes the software used by the origin server that handled the request — that is, the server that generated the response`, as I interpret that, it means that when proxying, you should proxy the `Server` header from the origin aswell, since the proxied server is what is actually generating the response.

Reply:
Right. I didn't consider that.

Reply:
In the latest commit, I've disabled the ability to override the `Connection` header and left comments elaborating on that + why `TryAdd` was used